streams/src/main/java/org/apache/kafka/streams/processor/internals/PartitionGroup.java

```
@@ -84,6 +84,27 @@ RecordQueue queue() {
         streamTime = RecordQueue.UNKNOWN;
     }
 
+    long timestamp() {
+        return streamTime;
+    }
+
+    long getPartitionTimestamp(final TopicPartition partition) {
+        if (partitionQueues.get(partition) == null) {
+            return RecordQueue.UNKNOWN;
+        }
+        return partitionQueues.get(partition).partitionTime();
+    }
+
+    void setPartitionTimestamp(final TopicPartition partition, final long timestamp) {
+        if (partitionQueues.get(partition) == null) {
+            return;
```

Similar as above, we should throw an exception here.

---

Ah, but if we do throw a NullPointer here, the test fails. So I don't know if that is what we really should do.

---

Not sure -- but maybe you setup the test incorrectly? `PartitionGroup` constructor get a `Map<TopicPartition, RecordQueue> partitionQueues` and you should only call `setPartitionTimestamp()` for `TopicPartitions` that are provided by this map. Could this explain the test issue?

---

Oh, that might be the case. I was going through the consumer assignment's topic partitions instead. Will check it out.